botocore/client.py

```diff
@@ -651,36 +651,41 @@ def can_paginate(self, operation_name):
         actual_operation_name = self._PY_TO_OP_NAME[operation_name]
         return actual_operation_name in self._cache['page_config']
 
-    def _get_waiter_config(self):
-        if 'waiter_config' not in self._cache:
-            try:
-                waiter_config = self._loader.load_service_model(
-                    self._service_model.service_name,
-                    'waiters-2',
-                    self._service_model.api_version)
-                self._cache['waiter_config'] = waiter_config
-            except DataNotFoundError:
-                self._cache['waiter_config'] = {}
-        return self._cache['waiter_config']
-
-    def get_waiter(self, waiter_name):
-        config = self._get_waiter_config()
-        if not config:
-            raise ValueError("Waiter does not exist: %s" % waiter_name)
-        model = waiter.WaiterModel(config)
+    @CachedProperty
+    def _waiter_config(self):
+        try:
+            return self._loader.load_service_model(
+                self._service_model.service_name,
+                'waiters-2',
+                self._service_model.api_version)
+        except DataNotFoundError:
+            return {}
+
+    def get_waiter(self, waiter_name, **config):
```

Comment:
I would prefer to be explicit here rather than accepting any keyword arguments. Specifically we should only allow setting `delay` and `max_attempts` for now.